include/inviwo/core/util/stdextensions.h

```diff
@@ -577,6 +581,30 @@ constexpr bool is_callable_with(F&&) {
     return std::is_invocable_v<F, A...>;
 }
 
+/**
+ * Get the index of a type in a tuple, returns the index of the first matching type
+ */
+template <class T, typename Tuple, size_t count = 0>
+constexpr size_t index_of() {
+    if constexpr (std::is_same_v<T, std::tuple_element_t<count, Tuple>>) {
+        return count;
+    } else {
+        return index_of<T, Tuple, count + 1>();
```

Comment:
What happens if there is no such type? Does it generate a compile error?

Reply:
That or the compiler will exceed its recursion limit when we get to -1...